### ISSUE TITLE
🐛[BUG] EditableProTable rows don't have unique keys in `name` mode

### ISSUE THREAD

**Issue by zhxnlai** [ant-design/pro-components#5265]
#### Context
In `name` mode, EditableProTable [assigns an index](https://github.com/ant-design/pro-components/blob/d882705b6c2e98e6261e981509b24a62ad66b56c/packages/table/src/components/EditableTable/index.tsx#L178
) to each row. `editableUtils.actionRender` uses row index instead of row key to edit and delete rows. Here is [an example test](https://github.com/ant-design/pro-components/blob/d882705b6c2e98e6261e981509b24a62ad66b56c/tests/table/editor-table-two.test.tsx#L892).

#### Problem
This works for flat tables but not for nested tables. In the current implementation, when a table has nested rows, child rows have the same indexes as their parents. This means row indexes are no longer unique. Because child rows have duplicate keys, they are not editable or deletable. 

As illustrated below, child rows have the same indexes as their parents:

```javascript
const defaultData: DataSourceType[] = [
  {
    id: 624748504, // Row index is 0.
    title: '活动名称一',
    decs: '这个活动真好玩',
    state: 'open',
    created_at: '2020-05-26T09:42:56Z',
    update_at: '2020-05-26T09:42:56Z',
    children: [
      {
        id: 6246912293, // Row index is 0.
        title: '活动名称二',
        decs: '这个活动真好玩',
        state: 'closed',
        created_at: '2020-05-26T08:19:22Z',
        update_at: '2020-05-26T08:19:22Z',
      },
    ],
  },
  {
    id: 624691229, // Row index is 1.
    title: '活动名称二',
    decs: '这个活动真好玩',
    state: 'closed',
    created_at: '2020-05-26T08:19:22Z',
    update_at: '2020-05-26T08:19:22Z',
  },
];
```

### 💻 复现代码

The easiest way to reproduce this is to look at [this test](https://github.com/ant-design/pro-components/blob/d882705b6c2e98e6261e981509b24a62ad66b56c/tests/table/editor-table-two.test.tsx#L823). The problem arises when we change the test to edit a nested row, `6246912293` (child of `624748504`).  There is no easy way to refer to row `6246912293` by row index.
```diff
  it('📝 support onValueChange when has name', async () => {
    const onValueChangeFn = jest.fn();
    const actionRef = React.createRef<ActionType | undefined>();
    const wrapper = render(
      <ProForm
        initialValues={{
          table: defaultData,
        }}
      >
        <EditableProTable<DataSourceType>
          rowKey="id"
          actionRef={actionRef as any}
+         // Note that this table is in name mode.
          name="table"
          pagination={{
            pageSize: 2,
          }}
          editable={{
            onValuesChange: (record) => {
              onValueChangeFn(record.id);
            },
          }}
          toolBarRender={(action) => [
            <a
              key="edit"
              id="start"
              onClick={() => {
+               action?.startEditable('6246912293');
-               action?.startEditable('624748504');
              }}
            >
              开始编辑
            </a>,
            <a
              key="end"
              id="end"
              onClick={() => {
+               action?.startEditable('6246912293');
-               action?.cancelEditable('624748504');
              }}
            >
              结束编辑
            </a>,
          ]}
          columns={columns}
        />
      </ProForm>,
    );
    await waitForComponentToPaint(wrapper, 1200);

    act(() => {
      wrapper.queryByText('开始编辑')?.click();
    });
    await waitForComponentToPaint(wrapper, 1000);

    act(() => {
      fireEvent.change(
        wrapper.container
          .querySelectorAll('.ant-table-tbody tr.ant-table-row')[0]
          .querySelectorAll('td .ant-input')[0],
        {
          target: {
            value: 'qixian',
          },
        },
      );
    });

    await waitForComponentToPaint(wrapper, 100);
    expect(onValueChangeFn).toBeCalledWith(624748504);

    act(() => {
+     // Row 6246912293 does not have a unique index. 
+     actionRef.current?.cancelEditable(???);
      actionRef.current?.cancelEditable(0);
    });
    await waitForComponentToPaint(wrapper, 1000);
    expect(
      wrapper.container
        .querySelectorAll('.ant-table-tbody tr.ant-table-row')[0]
        .querySelectorAll('td .ant-input').length,
    ).toBe(0);
  });

```

### © 版本信息

- ProComponents 版本: @ant-design/pro-table@2.74.2
- 开发环境 mac OS

**Comment by Dunqing** [comment]
Did you fix it in #5244?

**Comment by Dunqing** [comment]
It looks like depth+index can solve this issue

**Comment by zhxnlai** [comment]
> Did you fix it in #5244?

I didn't. The fix seems to require a big change.

**Comment by chenshuai2144** [comment]
> TAutomatically replying with ChatGPT can be attempted, but it cannot be guaranteed to be completely accurate and may not fully address all issues. Please feel free to reply if you have any further questions or concerns.
 > 此回复基于 ChatGPT 自动生成，可以尝试下方案，官方人员会在一定时间后继续继续处理。


谢谢您的回复！我会考虑给您们提 PR。如果我有任何问题，我会向您们寻求帮助。非常感谢您们对社区的贡献！
